views/js/list-comments.js

```diff
@@ -59,31 +65,80 @@ jQuery(document).ready(function () {
-      if (jsonResponse.comments && jsonResponse.comments.length > 0) {
-        populateComments(jsonResponse.comments);
-        if (jsonResponse.comments_nb > jsonResponse.comments_per_page) {
-          $('#product-comments-list-pagination').pagination({
-            currentPage: page,
-            items: jsonResponse.comments_nb,
-            itemsOnPage: jsonResponse.comments_per_page,
-            cssStyle: '',
-            prevText: '<i class="material-icons" data-icon="chevron_left"></i>',
-            nextText: '<i class="material-icons" data-icon="chevron_right"></i>',
-            useAnchors: false,
-            displayedPages: 2,
-            onPageClick: paginateComments
-          });
-        } else {
-          $('#product-comments-list-pagination').hide();
-        }
-      } else {
-        commentsList.html('');
-        emptyProductComment.show();
-        commentsList.append(emptyProductComment);
+  async function fetchComments(page) {  
+    let response = await fetch(commentsListUrl + "&page=" + page);
+
+    if (response.status === 200) {
+        let data = await response.text();
+        populateComments((JSON.parse(data)).comments);
+    }
+  }
+  
+  $(pagesListId + ' li').on('click',
+    function() {
+      let oldCount = commentsList.data('current-page');
+      let newCount = $(this).data('page');
+      
+      if (newCount === prevCount) {  // click prev
+        newCount = oldCount - 1;
+        if (newCount <= 0) return;
       }
-    });
+      if (newCount === nextCount) { // click next
+        newCount = oldCount + 1;
+        if (newCount >= nextCount) return;
+      }
+      
+      $(pageIdPrefix + oldCount + ' span').removeClass('current');
+      $(pageIdPrefix + oldCount).removeClass('active');
+      
+      fetchComments(newCount); // fetch new page's comments                  
+
+      $(pageIdPrefix + newCount).addClass('active');
+      $(pageIdPrefix + newCount + ' span').addClass('current');
+      
+      $(pageIdPrefix + newCount + ' span').html(newCount);
+      commentsList.data('current-page', newCount);      
+
+      if (newCount === 1) // disable prev
+        $(pageIdPrefix + prevCount).addClass('disabled');
+      else   
+        $(pageIdPrefix + prevCount).removeClass('disabled');
+
+      if (newCount === totalPages)  // disable next
+        $(pageIdPrefix + nextCount).addClass('disabled');
+      else
+        $(pageIdPrefix + nextCount).removeClass('disabled');
+
+      // long list with over 9 pages      
+      if (9 <= totalPages) {        
+        generateGap(newCount, prevCount);
+        generateGap(newCount, nextCount);
+      }
+    }
+  )
+
+  function generateGap(start, stop) {
+    if (start == stop)
+      return 0;
+    let step = (start < stop) ? +1 : -1;
+    let i = start + step;  
+    if (4 < Math.abs(stop - start)) {
+      $(pageIdPrefix + i).removeClass('hidden').removeClass('disabled');
+      $(pageIdPrefix + i + ' span').html(i);
```

Comment:
Consider using template literals instead of contatenating with `+` (here and in other selectors when applicable)

```suggestion
      $(`${pageIdPrefix}${i} span`).html(i);
```

Reply:
Applied. We have a bunch of $$$ now :bank: